pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -607,7 +607,7 @@ def select_training_type_plugin(self) -> TrainingTypePlugin:
             use_torchelastic_ddp = self.use_ddp and TorchElasticEnvironment.is_using_torchelastic()
             use_kubeflow_ddp = self.use_ddp and KubeflowEnvironment.is_using_kubeflow()
             use_ddp_spawn = self._distrib_type == DistributedType.DDP_SPAWN
-            use_ddp_cpu_spawn = self.use_ddp and self.use_cpu
+            use_ddp_cpu_spawn = self._distrib_type == DistributedType.DDP_SPAWN and self.use_cpu
```

Comment:
I think it is in multiple places... when I decoupled it before... :)

Reply:
Updated!